text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,467 @@
+```
+
+For reference: the output of this command can be found [here](https://gist.github.com/eladb/bf417c07444027d6954360295df4ee37#file-awsqs-vpc-vpcqs-module-json). 
+The parsed `Schema` field can be found [here](https://gist.github.com/eladb/bf417c07444027d6954360295df4ee37#file-awsqs-vpc-vpcqs-module-schema-json).
+
+Now, we need to filter all non-`AWS::` types (the `AWS::` types are actually L1s), and then generate 
+an L1 construct for each one. This includes:
+
+1. Generating the construct class that extends `CfnResource`. I think we can use submodules to 
+   represent the type namespace (`awsqs.eks` in this case), and the name of the resource as the construct name `Cluster`.
+2. Generate an `XxxProps` JSII struct for this construct based on the `Schema` field. 
+   This can be done using [`json2jsii`](https://github.com/cdklabs/json2jsii), which is the same 
+   tool we use to implemented the `import` command in CDK8s and CDKtf.
+
+
+### Is this a breaking change?
+
+Nope.
+
+### What are the drawbacks of this solution?
+
+> Describe any problems/risks that can be introduced if we implement this RFC.
+
+### What alternative solutions did you consider?
+
+#### `cdk import` versus library
+
+As mentioned above, an alternative approach would be to add support for `import` in the CDK CLI. The downsides of this approach (at this point) 
+are that (a) it is a bigger investment as it involves a change to the CDK CLI; and (b) the Construct Hub won't be able to surface these
+types automatically (we will need some custom support for "imported" types, which is something that we would like to add to the Construct Hub
+in the future for all CDK domains).
+
+#### Monolithic versus module per resource
+
+As mentioned above, due to the fact that resource schemas are versioned and we want to allow users to select which
+version of the resource they use, we determined that the best approach is to publish a module for each resource.
+This has logicical/operational implications that we believe can mostly be mitigated using projen's capabilities to
+reason about the project structure and publishing workflows using code.
+
+### What is the high level implementation plan?
+
+- [ ] Implement [`cdk-import`](https://github.com/cdklabs/cdk-import) as a command-line tool that generates L1 constructs for registry extensions.
+- [ ] Create a mono-repo-style project using projen that utilizes `cdk-import` to generate a module for each registry resource.
+
+
+### Are there any open issues that need to be addressed later?
+
+* __Property name conversion__: `json2jsii` converts field names to camelCase to adhere with naming typescript naming conventions. 
+  This means that we need to map field names back when we define the `CfnResource`. I think we might want to add a 
+  feature to `json2jsii` that will generate conversion functions that can be used to convert back data types to the original schema.
+
+## Appendix
+
+```ts
+export class Cluster extends CfnResource {
```

Comment:
Do we call this `CfnCluster` to indicate it's a raw CFN resource with no human flavoring?

What do we do for attributes? `attrClusterName` like L1s? And mutable properties like L1s? People are going to need these.

At the same time, having both `cluster.attrClusterName` and `cluster.clusterName` available has led a LOT of people to use the wrong property on our current collection of L1s. `cluster.propClusterName` instead maybe?

Reply:
Deleted the appendix - we can discuss the details over https://github.com/cdklabs/cdk-import/pull/2